tests/unit/test_normalize_icon_type.py

```diff
@@ -0,0 +1,67 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2022, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+import os
+import sys
+from pathlib import Path
+
+import pytest
+
+import PyInstaller
+
+
+def test_normalize_icon_type(monkeypatch, tmp_path):
```

Comment:
I think the single test should be split into multiple separate tests, so that the non-Pillow ones get marked as passed instead of skipped when Pillow is unavailable:
- file-not-found error test
- native image
- converted image
- random image
- install-the-pillow error

Reply:
Oh, I misread this. I separated the tests out into two groups, one of which needs PIL and one of which doesn't.

Reply:
I'd prefer to see them separate, but it's not a deal-breaker at this point. The current split does accomplish the objective, and we can split them up later if necessary.